File: 프로그래머스/lv2/42842. 카펫/카펫.js
function solution(brown, yellow) { // 10, 2
    const answer = [0, 0];
    const S = brown + yellow; // 전체 면적 S = 12
    
    for (let width = S; width > 0; width--) { // width = 4
        if (S % width !== 0) continue; // 전체 면적이 너비로 나누어떨어지지 않는 경우는 스킵
        
        const height = S / width;                   // height = 3
        const yellowS = (width - 2) * (height - 2); // 2 * 1 = 2
        const brownS = S - yellowS;                 // 10
        
        if (yellowS === yellow && brownS === brown) { // 조건이 맞아 떨어지면
            answer[0] = width;
            answer[1] = height;
            break; // 반복문 탈출
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow): # 10, 2
    total_surface = brown + yellow # 12
    
    # total_surface의 약수를 구한다.
    possible_matches = [] # [[12, 1], [6, 2], [4, 3]]
    for i in range(1, int(total_surface ** 0.5) + 1):
        if total_surface % i == 0:
            possible_matches.append([total_surface // i, i])
    
    for w, h in possible_matches:
        if (w - 2) * (h - 2) == yellow:
            return [w, h]
    